realm/realm-library/src/syncIntegrationTest/java/io/realm/objectserver/ManagementRealmTests.java

```diff
@@ -51,7 +53,42 @@
     @Rule
     public RunInLooperThread looperThread = new RunInLooperThread();
 
-    @Ignore("TODO: Test is currently flaky. See https://github.com/realm/realm-java/pull/4066")
+    // This is primarily a test making sure that an admin user actually connects correctly to ROS.
+    // See https://github.com/realm/realm-java/issues/4750
+    @Test
+    @RunTestInLooperThread
+    public void adminUser_writeInvalidPermissionOffer() {
+        final SyncUser user = UserFactory.createAdminUser(Constants.AUTH_URL);
+        assertTrue(user.isValid());
+        Realm realm = user.getManagementRealm();
+        looperThread.closeAfterTest(realm);
+        looperThread.runAfterTest(new Runnable() {
+            @Override
+            public void run() {
+                user.logout();
+            }
+        });
+        realm.beginTransaction();
+        // Invalid Permission offer
+        realm.copyToRealm(new PermissionOffer("*", true, true, false, null));
```

Comment:
it's outside the scope of this PR, but is it possible to refactor `PermissionOffer` to use a builder pattern? reading three boolean, can't know which permission they affect